Failure on travis: waitActive time-outs

`test_wait_window` for `waitActive` on the `py34-qt5` environment on Travis fails with:

```
=================================== FAILURES ===================================
______________________ test_wait_window[waitActive-True] _______________________
show = True, method_name = 'waitActive'
qtbot = <pytestqt.qtbot.QtBot object at 0x7ff758477780>
    @pytest.mark.parametrize('show', [True, False])
    @pytest.mark.parametrize('method_name', ['waitExposed', 'waitActive'])
    def test_wait_window(show, method_name, qtbot):
        """
        Using one of the wait-widget methods should not raise anything if the widget
        is properly displayed, otherwise should raise a TimeoutError.
        """
        method = getattr(qtbot, method_name)
        if qt_api.pytest_qt_api != 'pyqt5':
            with pytest.raises(RuntimeError) as exc_info:
                with method(None, None):
                    pass
            assert str(exc_info.value) == 'Available in PyQt5 only'
        else:
            widget = qt_api.QLineEdit()
            qtbot.add_widget(widget)
            if show:
                with method(widget, timeout=1000):
                    widget.move(100, 100)
                    widget.resize(100, 100)
                    widget.setFocus()
>                   widget.show()
E                   pytestqt.qtbot.TimeoutError: widget <PyQt5.QtWidgets.QLineEdit object at 0x7ff7584cb828> not activated in 1000 ms.
```

It was tried to move and resize the window based on the docs for `qWaitForWindowActive`:

```c++
    inline static bool qWaitForWindowActive(QWindow *window, int timeout = 5000)
    {
        QElapsedTimer timer;
        timer.start();
        while (!window->isActive()) {
            int remaining = timeout - int(timer.elapsed());
            if (remaining <= 0)
                break;
            QCoreApplication::processEvents(QEventLoop::AllEvents, remaining);
            QCoreApplication::sendPostedEvents(0, QEvent::DeferredDelete);
            QTest::qSleep(10);
        }
        // Try ensuring the platform window receives the real position.
        // (i.e. that window->pos() reflects reality)
        // isActive() ( == FocusIn in case of X) does not guarantee this. It seems some WMs randomly
        // send the final ConfigureNotify (the one with the non-bogus 0,0 position) after the FocusIn.
        // If we just let things go, every mapTo/FromGlobal call the tests perform directly after
        // qWaitForWindowShown() will generate bogus results.
        if (window->isActive()) {
            int waitNo = 0; // 0, 0 might be a valid position after all, so do not wait for ever
            while (window->position().isNull()) {
                if (waitNo++ > timeout / 10)
                    break;
                qWait(10);
            }
        }
        return window->isActive();
    }

``` 

But so far no good.

--- a/.travis.yml
+++ b/.travis.yml
@@ -23,8 +23,8 @@ env:
 install:
  - sudo apt-get update
 
- # Xvfb
- - sudo apt-get install -y xvfb
+ # Xvfb / window manager
+ - sudo apt-get install -y xvfb herbstluftwm
 
  # Qt
  - python scripts/install-qt.py
@@ -38,10 +38,16 @@ install:
  # others
  - pip install coveralls --use-wheel
 
+before_script:
+ - "export DISPLAY=:99.0"
+ - "sh -e /etc/init.d/xvfb start"
+ - sleep 3 # give xvfb some time to start
+ - "herbstluftwm &"
+ - sleep 1
+
 script:
  - python setup.py develop
- # change screen size and depth in attempt to fix waitActive test (#160)
- - catchsegv xvfb-run --server-args="-screen 0 640x480x24" coverage run --source=pytestqt setup.py test
+ - catchsegv coverage run --source=pytestqt setup.py test
  - tox -e lint
 
 after_success:

--- a/pytestqt/qt_compat.py
+++ b/pytestqt/qt_compat.py
@@ -106,7 +106,6 @@ class _QtApi:
             self.Property = QtCore.Property
             self.QApplication = QtGui.QApplication
             self.QWidget = QtGui.QWidget
-            self.QLineEdit = QtGui.QLineEdit
             self.QStringListModel = QtGui.QStringListModel
             self.qInstallMsgHandler = QtCore.qInstallMsgHandler
 
@@ -137,7 +136,6 @@ class _QtApi:
                 _QtWidgets = _import_module('QtWidgets')
                 self.QApplication = _QtWidgets.QApplication
                 self.QWidget = _QtWidgets.QWidget
-                self.QLineEdit = _QtWidgets.QLineEdit
                 self.qInstallMessageHandler = QtCore.qInstallMessageHandler
 
                 self.QStringListModel = QtCore.QStringListModel

--- a/tests/test_basics.py
+++ b/tests/test_basics.py
@@ -89,8 +89,6 @@ def test_wait_window(show, method_name, qtbot):
         if show:
             with method(widget, timeout=1000):
                 widget.show()
-                if 'TRAVIS' in os.environ and method_name == 'waitActive':
-                    pytest.xfail('skipping this check on travis, see #160')
         else:
             with pytest.raises(qtbot.TimeoutError):
                 with method(widget, timeout=100):

--- a/tests/test_basics.py
+++ b/tests/test_basics.py
@@ -84,7 +84,7 @@ def test_wait_window(show, method_name, qtbot):
                 pass
         assert str(exc_info.value) == 'Available in PyQt5 only'
     else:
-        widget = qt_api.QLineEdit()
+        widget = qt_api.QWidget()
         qtbot.add_widget(widget)
         if show:
             with method(widget, timeout=1000):

--- a/tests/test_basics.py
+++ b/tests/test_basics.py
@@ -88,9 +88,6 @@ def test_wait_window(show, method_name, qtbot):
         qtbot.add_widget(widget)
         if show:
             with method(widget, timeout=1000):
-                widget.move(100, 100)
-                widget.resize(100, 100)
-                widget.setFocus()
                 widget.show()
                 if 'TRAVIS' in os.environ and method_name == 'waitActive':
                     pytest.xfail('skipping this check on travis, see #160')

